README.md

```diff
@@ -66,9 +66,20 @@ followed by a possibly unbounded number of `onNext` signals (as requested by `Su
 #### NOTES
 
 - The specifications below use binding words in capital letters from https://www.ietf.org/rfc/rfc2119.txt
-- The terms `emit`, `signal` or `send` are interchangeable. The specifications below will use `signal`.
-- The terms `synchronously` or `synchronous` refer to executing in the calling `Thread`.
-- The term "return normally" means "only throws exceptions that are explicitly allowed by the rule".
+
+### Glossary
+
+| Term                      | Definition                                                                                             |
+| ------------------------- | ------------------------------------------------------------------------------------------------------ |
+| <a name="term_signal">Signal</a> | As a noun: one of the `onSubscribe`, `onNext`, `onComplete` or `onError` methods. As a verb: calling/invoking a signal. |
+| <a name="term_demand">Demand</a> | As a noun, the aggregated number of elements requested by a Subscriber which is yet to be delivered (fulfilled) by the Publisher. As a verb, the act of `request`-ing more elements. |
+| <a name="term_sync">Synchronous(ly)</a> | Executes on the calling Thread. |
+| <a name="term_return_normally">Return normally</a> | Returns a value of the declared type alternatively throws an exception permitted by this specification. The only legal way to signal failure to a `Subscriber` is via the `onError` method.|
+| <a name="term_responsivity">Responsivity</a> | Readiness/ability to respond. In this document used to indicate that the different components should not impair eachothers ability to respond. |
+| <a name="term_non-obstructing">Non-obstructing</a> | Quality describing a method which is as quick to execute as possible—on the calling thread. This means, for example, avoids heavy computations and other things that would stall the caller´s thread of execution. |
+| <a name="term_terminal_state">Terminal state</a> | For a Publisher: The point where `onComplete` or `onError` has been signalled. For a Subscriber: The point where an `onComplete` or `onError` has been received.|

```

Comment:
A `cancel` puts the `Subscriber`+`Subscription` pair into a terminal state. It's confusing from the perspective of a `Publisher`, as in a stateless `Publisher`, the `Publisher` isn't doing anything, it's all the `Subscription`. Yet we say in this description of "terminal state" that it puts the `Publisher` in a terminal state ... which is somewhat meaningless unless it is stateful and "hot". Hence the confusion with this description. 

Seems to me that what is trying to be said is:

- `Subscriber` is in a terminal state when `onComplete` or `onError` are signaled, or when it sends `cancel` to its `Subscription` (acknowledging the async race). 
- `Subscription` is in a terminal state when it receives `cancel` or the `onComplete` or `onError` signals are sent for that `Subscription`+`Subscriber` pair.
- A stateless `Publisher` has no terminal state.
- A stateful `Publisher` is in a terminal state when it sends `onComplete` or `onError`, but `cancel` does not necessarily affect it (it can "pause" itself though, such as when doing `refCount`)